Shared/Network/Resource.swift

```diff
@@ -0,0 +1,48 @@
+public typealias JSONDictionary = [String: AnyObject]
+
+public struct Resource<T> {
+
+    public let url: URL
+    public let parse: (Data) -> T?
+
+    public init(url: URL, parseJSON: @escaping (Any) -> T?) {
+        self.url = url
+        self.parse = { data in
+            let json = try? JSONSerialization.jsonObject(with: data, options: [])
+            return json.flatMap(parseJSON)
+        }
+    }
+
+}
+
+public struct Resources {
+
+    private static let baseUrl = "https://api.themoviedb.org/3"
+    private static let apiKey = "1f54bd990f1cdfb230adb312546d765d"
+    private static let imageBaseUrl = "https://image.tmdb.org/t/p/w500"
+
+    static func upcomingMovies(page: Int = 1) -> Resource<[Movie]> {
+        let upcomingMoviesUrl = URL(string: "\(baseUrl)/movie/upcoming?api_key=\(apiKey)&page=\(page)")!
```

Comment:
This API is very strange. I suggest to ArcTouch to use the www.tvmaze.com API. It is mor consistent.